src/chunk_insert_state.c

```diff
@@ -204,6 +205,168 @@ create_chunk_result_relation_info(ChunkDispatch *dispatch, Relation rel, Index r
 	return rri;
 }
 
+static ProjectionInfo *
+get_adjusted_projection_info_returning(ProjectionInfo *orig, List *returning_clauses, AttrNumber *map,
+									   int map_size, Index varno, Oid rowtype,
+									   TupleDesc chunk_desc)
+{
+	bool		found_whole_row;
+
+	Assert(returning_clauses != NIL);
+
+	/* map hypertable attnos -> chunk attnos */
+	returning_clauses = (List *) map_variable_attnos_compat((Node *) returning_clauses, varno, 0, map, map_size,
+															rowtype, &found_whole_row);
+
+	return ExecBuildProjectionInfoCompat(returning_clauses, orig->pi_exprContext,
+										 get_projection_info_slot_compat(orig), NULL, chunk_desc);
+}
+
+static ExprState *
+get_adjusted_onconflictupdate_where(ExprState *hyper_where_state, List *where_quals, AttrNumber *map,
+									int map_size, Index varno, Oid rowtype)
+{
+	bool		found_whole_row;
+
+	Assert(where_quals != NIL);
+	/* map hypertable attnos -> chunk attnos for the hypertable */
+	where_quals = (List *) map_variable_attnos_compat((Node *) where_quals, varno, 0, map, map_size,
+													  rowtype, &found_whole_row);
+	/* map hypertable attnos -> chunk attnos for the "excluded" table */
+	where_quals = (List *) map_variable_attnos_compat((Node *) where_quals, INNER_VAR, 0, map, map_size,
+													  rowtype, &found_whole_row);
+#if PG96
+	return ExecInitExpr((Expr *) where_quals, NULL);
+#else
```

Comment:
Do we want an `#elif PG10` here? Otherwise we're saying the else-clause is for all other versions. We should probably be explicit about support...